File: shared/test-utils/js/fixtures/fixture-loader.js
/**
 * Fixture loader - loads JSON test fixtures from shared/fixtures/
 */

const fs = require("fs");
const path = require("path");

/**
 * Load a fixture by spec ID and variant
 *
 * @param {string} specId - The spec ID (e.g., "1-simple", "2-simple-with-error")
 * @param {string} variant - The fixture variant (e.g., "agentic", "low-level")
 * @returns {Object} The parsed fixture object
 * @throws {Error} If fixture file not found
 */
function loadFixture(specId, variant = "agentic") {
  // Fixtures are in shared/specs/{specId}/fixture-{variant}.json
  const fixturePath = path.join(__dirname, "../../../specs", specId, `fixture-${variant}.json`);

  if (!fs.existsSync(fixturePath)) {
    throw new Error(`Fixture not found: ${specId} (variant: ${variant}) at ${fixturePath}`);
  }

  const content = fs.readFileSync(fixturePath, "utf-8");
  return JSON.parse(content);
}

module.exports = {
  loadFixture,
};

File: shared/test-utils/py/fixtures/fixture_loader.py
"""
Fixture loader - loads JSON test fixtures from shared/fixtures/
"""

import json
import os
from pathlib import Path
from typing import Dict, Any


def load_fixture(spec_id: str, variant: str = "agentic") -> Dict[str, Any]:
    """
    Load a fixture by spec ID and variant

    Args:
        spec_id: The spec ID (e.g., "1-simple", "2-simple-with-error")
        variant: The fixture variant (e.g., "agentic", "low-level")

    Returns:
        The parsed fixture object

    Raises:
        FileNotFoundError: If fixture file not found
    """
    # Fixtures are in shared/specs/{spec_id}/fixture-{variant}.json
    current_dir = Path(__file__).parent
    fixture_path = current_dir / "../../../specs" / spec_id / f"fixture-{variant}.json"
    fixture_path = fixture_path.resolve()

    if not fixture_path.exists():
        raise FileNotFoundError(f"Fixture not found: {spec_id} (variant: {variant}) at {fixture_path}")

    with open(fixture_path, "r", encoding="utf-8") as f:
        return json.load(f)
